zig panic while linking mimalloc test

### Zig Version

0.11.0-dev.479+f68bfe2eb

### Steps to Reproduce and Observed Behavior

1. Download/clone mimalloc (https://github.com/microsoft/mimalloc)
2. mkdir build; cd build; CC="zig cc" CXX="zig C++" cmake ..
3. make

```
[ 86%] Linking C executable mimalloc-test-api
LLD Link... thread 171188 panic: attempt to use null value
???:?:?: 0x167510e in linkWithLLD (/home/rganesan/git/zig/src/zig_llvm.cpp)
???:?:?: 0x1490126 in flush (/home/rganesan/git/zig/src/zig_llvm.cpp)
???:?:?: 0x128205b in flush (/home/rganesan/git/zig/src/zig_llvm.cpp)
???:?:?: 0x12818c5 in flush (/home/rganesan/git/zig/src/zig_llvm.cpp)
???:?:?: 0x1285440 in update (/home/rganesan/git/zig/src/zig_llvm.cpp)
???:?:?: 0x12b35ad in updateModule (/home/rganesan/git/zig/src/zig_llvm.cpp)
???:?:?: 0x11b7a72 in buildOutputType (/home/rganesan/git/zig/src/zig_llvm.cpp)
???:?:?: 0x119682f in mainArgs (/home/rganesan/git/zig/src/zig_llvm.cpp)
???:?:?: 0x13179bf in main (/home/rganesan/git/zig/src/zig_llvm.cpp)
Error running link command: Subprocess aborted
make[2]: *** [CMakeFiles/mimalloc-test-api.dir/build.make:100: mimalloc-test-api] Error 1
make[1]: *** [CMakeFiles/Makefile2:171: CMakeFiles/mimalloc-test-api.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```
Consistently reproducible on Ubuntu 22.10

### Expected Behavior

Linking should not panic

--- a/src/link/Elf.zig
+++ b/src/link/Elf.zig
@@ -1679,7 +1679,7 @@ fn linkWithLLD(self: *Elf, comp: *Compilation, prog_node: *std.Progress.Node) !v
             }
             for (self.base.options.objects) |obj| {
                 if (Compilation.classifyFileExt(obj.path) == .shared_library) {
-                    const lib_dir_path = std.fs.path.dirname(obj.path).?;
+                    const lib_dir_path = std.fs.path.dirname(obj.path) orelse continue;
                     if ((try rpath_table.fetchPut(lib_dir_path, {})) == null) {
                         try argv.append("-rpath");
                         try argv.append(lib_dir_path);

